### PR TITLE
Reporttool

### PR DESCRIPTION
**Pull request by hxiedwr**
I am merging the ReportTool branch to wrims-devops branch. Please review and approve it.

Please make sure what I am doing is merging ReportTool branch to wrims-devops branch but not the other way around.

### CODE REVIEW

File: wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java [hunk 1]
```diff
@@ -198,12 +196,40 @@ public void doProcessing(IProgressMonitor monitor) {
 		monitor.worked(20);
 		
 		// open files 1 and file 2 and loop over to plot
-		Group dssGroupBase = Utils.opendss(scalars.get("FILE_BASE"));
-		Group dssGroupAlt = Utils.opendss(scalars.get("FILE_ALT"));
+		HecTimeSeries htsBase = new HecTimeSeries(scalars.get("FILE_BASE"));
```
Comment (rma-tom) [inline wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java:199]: If there's a console output for messages, consider setting the DSS messaging level lower than the default, for example
		HecTimeSeries.setMessageLevel(3);

File: wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java [hunk 2]
```diff
@@ -249,61 +273,69 @@ public void doProcessing(IProgressMonitor monitor) {
...
+				tscBase=Utils.cfs2taf(tscBase);
+				tscAlt=Utils.cfs2taf(tscAlt);
+			} else if (pathMap.units.equalsIgnoreCase("TAF2CFS")) {
+				tscBase=Utils.taf2cfs(tscBase);
+				tscAlt=Utils.taf2cfs(tscAlt);
 			}
-			String data_units = Utils.getUnits(refBase, refAlt);
-			String data_type = Utils.getType(refBase, refAlt);
+			String data_units = tscBase.units;
+			String data_type = tscBase.type;
```
Comment (rma-tom) [inline wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java:299]: TimeSeriesContainer.type is going to give you things like "PER-AVER" or "INST-VAL". You probably want to set your data_type from tscBase.parameter, which is going to be the C-part of the DSS path

**Review by rma-tom — CHANGES_REQUESTED** [review]

**Comment by hxiedwr** [comment]
Right now, we don’t have console outputs but haven’t got any complaints in the past. So I think this is not urgent.

The same goes to data_type, which is not used to generate tables or charts in the Report Tool.
________________________________

**Comment by rma-tom** [comment]
When I revised the CalLite report tool, the data_type was used to label the Y axes of the graphs, so I had graphs labeled with "PER-AVER" instead of "STORAGE" for example. If the axis labels in the WRIMS reports don't rely on the variable, then it's not important to make the change I suggested.

File: wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java (hunk 2 above)
Comment (rma-tom) [reply wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java:299]: per @hxiedwr "Right now, we don’t have console outputs but haven’t got any complaints in the past. So I think this is not urgent. "

Comment (rma-tom) [reply wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java:299]: Sorry -- that last comment belongs in the other location.

File: wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java (hunk 1 above)
Comment (rma-tom) [reply wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java:199]: per @hxiedwr "Right now, we don’t have console outputs but haven’t got any complaints in the past. So I think this is not urgent. "

File: wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java (hunk 2 above)
Comment (rma-tom) [reply wrims_v2/wrimsv2_plugin/src/wrimsv2_plugin/reporttool/Report.java:299]: Per @hxiedwr -- "The same goes to data_type, which is not used to generate tables or charts in the Report Tool."
If this won't cause problems with the labels of the axes, then it's not significant.

**Comment by hxiedwr** [comment]
Report Tool in WRIMS does not need type information to plot. Especially the type info is not always correct. We use "per-aver" for storage but actually should be "inst-val".

Basically we can think type is a dummy variable.




________________________________

**Review by rma-tom — COMMENTED** [review]
Are you able to run this in Eclipse? I tried, but the the re-arrangement of the directories for DevOps broke Eclipse projects. Maybe you can show me how. In code, though, this all looks good.

**Comment by hxiedwr** [comment]
So in CalLite Report Tool and EPPT, you may want to remove the type info from the axis.
________________________________

**Review by rma-tom — APPROVED** [review]
I hope that the switch from Vista to HecTimeSeries to read from DSS files will also make the reports run a bit faster.

**Comment by hxiedwr** [comment]
I am able to run. There is no compilation errors for wrims_plugin, hecdss, and jdiagram projects. The two xtext projects have compilation errors, which are hard to fix (it requires update to newer version of xtext, which may not work with the current code. So don't do it).

Two ways to work around:


  1.
launch the programs by ignoring the compilation errors in xtext projects.
  2.
copy the files under the two xtext projects from the master branch.

If you still have problem, let me know.




________________________________

**Review by rma-tom — DISMISSED** [review]
I submitted an approval, not sure why it's still blocked

**Comment by hxiedwr** [comment]
Because you were the last developer submitting the code change. GitHub required additional reviewer to approve it.



________________________________

**Review by adamkorynta — DISMISSED** [review]
Reviewed Tom's commit (did not review Hao's)

An issue should get created to consolidate this with the callite-report-tool library: https://github.com/CentralValleyModeling/callite-report-tool so that we don't have so much duplicated code.

**Pull request merged**